app/encryption.py

```diff
@@ -32,19 +32,70 @@ def init_app(self, app):
         self.serializer = URLSafeSerializer(app.config.get("SECRET_KEY"))
         self.salt = app.config.get("DANGEROUS_SALT")
 
-    def sign(self, to_sign):
-        return self.serializer.dumps(to_sign, salt=self.salt)
+    def _sign(self, to_sign: Any, salt: Optional[str] = None) -> str:
+        if salt is None:
+            salt = self.salt
+        return self.serializer.dumps(to_sign, salt=salt)
 
-    def verify(self, to_verify):
-        return self.serializer.loads(to_verify, salt=self.salt)
+    # NOTE: currently the verify checks agains the default salt as well as the salt passed in
+    # TODO: remove this once we've moved DANGEROUS_SALT to PASSWORD_SALT
+    def _verify(self, to_verify: str, salt: Optional[str] = None) -> Any:
+        if salt is None:
+            salt = self.salt
+        try:
+            return self.serializer.loads(to_verify, salt=salt)
+        except BadSignature:
+            return self.serializer.loads(to_verify, salt=self.salt)
+
+    # TODO: get rid of this after everything is signed with the new salts
+    # This is only needed where we look things up by the signed value:
+    #     - get_api_key_by_secret()
+    def sign_dangerous(self, to_sign: Any) -> str:
+        return self._sign(to_sign)
 
     def sign_notification(self, notification: NotificationDictToSign) -> SignedNotification:
         "A wrapper around the sign fn to define the argument type and return type"
-        return self.sign(notification)
+        return SignedNotification(self._sign(notification, "notification"))
 
     def verify_notification(self, signed_notification: SignedNotification) -> NotificationDictToSign:
         "A wrapper around the verify fn to define the argument type and return type"
-        return self.verify(signed_notification)
+        return self._verify(signed_notification, "notification")
+
+    def sign_personalisation(self, personalisation: dict) -> str:
```

Comment:
It seems like an opportunity to have a more OOP approach where we do not duplicate the methods, but rather have multiple instances of the signer, for each usage of signing where we'd provide the salts as a parameter upon instantiation via the constructor. Do you know what I mean?

Reply:
Practically too, that would minimize our required testing code to get full coverage if we only have the sign/verify methods instead of duplicating these (and every time we'd add a signer).

Reply:
👍 that's a really good idea! Makes code / tests cleaner and when / if we move towards different `SECRET_KEY`s for each it'll work with minimal code changes.